### ISSUE TITLE
Improve speed/rendering of web.threema.ch

### ISSUE THREAD

**Issue by ovalseven8** [threema-ch/threema-web#403]
Usually, I delete all cookies and the cache when I close the browser window. Over the time, I have noticed that loading `web.threema.ch` in my browser takes quite a long time for a website, also subjectively perceived it takes longer compared to competitors like `web.whatsapp.com`.

So, today I compared the speed using an online tool. I cannot say something about the methods used by this tool, but nonetheless it confirms my feelings about the long rendering time.

Here you can find the report:
https://www.dareboost.com/en/comparison/5a5b30e40cf2b528a72eaa86/5a5b30e40cf2b528a72eaa87/

Also a lot of things one can perhaps improve:
https://www.dareboost.com/en/report/5a5b30e40cf2b528a72eaa87

**Comment by joelfischerr** [comment]
Interestingly once caching is enabled `web.threema.ch` loads almost 3 times faster on my machines than `web.whatsapp.com` does, while with caching disabled `web.whatsapp.com` takes 4.5 seconds and `web.threema.ch` takes just a second more. I used Chromes developer tools to measure btw.

**Comment by dbrgn** [comment]
Currently there is no concatenation of JS libraries being done. With HTTP 2 multiplexing, reducing the number of requests (to the same host) should not really make a noticeable speed difference, but at the same time has the disadvantage of invalidating the entire cache even if just one library was updated. Furthermore, right now a user could verify that a library was distributed unmodified from the npm distribution, while that isn't possible anymore once concatenated.

We might introduce concatenation/minification of all libraries in the future, but I'm not sure about it yet.

@ovalseven8 were these tests done with HTTP 1 or 2?

**Comment by ovalseven8** [comment]
> @ovalseven8 were these tests done with HTTP 1 or 2?

That's just an external testing tool I found, so I can't guarantee, but the test probably used http/2.0.

At least they say:
> We have announced it a couple months ago: our web performance testing tool now fully supports HTTP/2 and our best practices repository has been updated to handle the particularities of this new protocol.

Source: https://blog.dareboost.com/en/2016/11/http2-changes-for-front-end-development/

**Comment by rugk** [comment]
Yeah just use http/2, I think others make no big difference than. But well … nobody verifies that by hand anyway.

**Comment by ovalseven8** [comment]
The rendering is starting very late for web.threema.ch
